src/constant_value.rs

```diff
@@ -30,6 +32,8 @@ pub enum ConstantValue {
     I128(i128),
     /// 64 bit floating point, stored as a u64 to make it comparable.
     F64(u64),
+    /// A string literal.
+    Str(String),
```

Comment:
A Str holding a String, well that should help reduce the confusion between String and str and &str :)

I would consider naming it StringLiteral(String), otherwise rust people are going to think of &str but see String and it just gets confusing. I guess that's what happens when your analyzer for a language is written in that language, though.

Reply:
I'll probably come back to this and find a way to make the parameter an actual &str. For now, the Str part strongly suggests the runtime type of the constant/literal and relates to the MIR data structures, in way that StringLiteral does not.